docs/knowledge_base/policy_examples.md

```diff
@@ -0,0 +1,185 @@
+
+```rego
+package policy
+
+import data.phylum.domain
+import data.phylum.level
+import future.keywords.contains
+import future.keywords.if
+import future.keywords.in
+
+issue contains "risk level cannot exceed medium" if {
+	data.issue.domain in {domain.AUTHOR, domain.ENGINEERING, domain.VULNERABILITY}
+	data.issue.severity > level.MEDIUM
+}
+
+issue contains "malicious risk level cannot exceed low" if {
+	data.issue.domain == domain.MALICIOUS
+	data.issue.severity > level.LOW
+}
+
+issue contains "license risk level cannot exceed high" if {
+	data.issue.domain == domain.LICENSE
+	data.issue.severity > level.HIGH
+}
+```
+
+Given the following input:
+
+```json
+{
+    "dependencies": [{
+        "ecosystem": "pypi",
+        "id": "07a3feb6-d6fb-523a-88f0-9896b86e6f93",
+        "issues": [{
+            "domain": "malicious",
+            "severity": 4,
+            "tag": "CM0004"
+        }],
+        "issues_complete": true,
+        "name": "example-package",
+        "version": "1.0.0"
+    }]
+}
+```
+
+When the policy fails, the output will look something like this:
+
+```json
+{
+  "dependencies": [
+    {
+      "errors": [],
+      "id": "07a3feb6-d6fb-523a-88f0-9896b86e6f93",
+      "issues": [
+        {
+          "reason": "malicious risk level cannot exceed low",
+          "tag": "CM0004"
+        }
+      ]
+    }
+  ],
+  "errors": []
+}
+```
+
+When Phylum sees this output from the policy, it will block the job and generate a report naming the package and describing the issue.
+
+## Blocking a dependency
+
+If a dependency is being blocked for some other reason, the policy can indicate this by using the `errors` property on the dependency in the output.
+
+The following policy blocks packages belonging to a namespace.
+
+```rego
+package policy
+
+import future.keywords.contains
+import future.keywords.if
+
+dependency contains "AGPL licensed software is not allowed." if {
+        regex.match("(?i)\\bAGPL\\b", data.dependency.license)
+}
+```
+
+Given the following input:
+
+```json
+{
+    "dependencies": [{
+        "ecosystem": "npm",
+        "id": "4cc36d79-b8ce-5b7d-89c1-6f6a31f59819",
+        "issues": [],
+        "issues_complete": true,
+        "license": "AGPL-3.0",
+        "name": "example-package",
+        "version": "1.0.0"
+    }]
+}
+```
+
+When the policy fails, the output will look something like this:
+
+```json
+{
+  "dependencies": [
+    {
+      "errors": [
+        "AGPL licensed software is not allowed."
+      ],
+      "id": "4cc36d79-b8ce-5b7d-89c1-6f6a31f59819",
+      "issues": []
+    }
+  ],
+  "errors": []
+}
+```
+
+When Phylum sees this output from the policy, it will block the job and generate a report naming the package and providing this message in the output.
+
+## Blocking a job
+
+Finally, if a policy wants to block a job for a reason that is not related to a specific dependency, the policy can provide errors at the top level.
```

Comment:
Similar to how dependencies are now blocked using `dependency` rules instead of setting errors, jobs are blocked using `job` rules.

Reply:
Updated